packages/typespec-autorest-canonical/src/emitter.ts

```diff
@@ -97,7 +97,7 @@ async function emitAllServices(
     const result = await getOpenAPIForService(context, options);
     const includedVersions = getVersion(program, service.type)
       ?.getVersions()
-      ?.map((item) => item.name);
+      ?.map((item) => item.value);
```

Comment:
the value could also be undefined need to `?? item.name`

Reply:
update the value